code/datums/extensions/assembly/assembly.dm

```diff
@@ -0,0 +1,122 @@
+/*
+ * Friendly class for managing component-level construction for /obj/items.
+ *
+ * Handles power, installation, and some processing.
+ * Basic usage:
+ *	1. Have Processing enabled on your item/machine. Call Process() on the assembly datum.
+ *	2. Call attackby on the assembly datum to handle tool interactions.
+ *	3. Recommended to also hook into ex_act, and any of the assembly_damage functions on base level for assembly to handle damage.
+ */
+
+/datum/extension/assembly
+	base_type = /datum/extension/assembly
+	var/assembly_name = "assembly"
+	var/list/parts = list()					// List of all components.
```

Comment:
Is it possible to LAZY these lists? This seems like an extension that will see a lot of use and there's at least three of them here.